feat: 🎸 littlefs as optional backend

peterbornerup opened the pull request.

optical-o commented:
First of all thank you for the cooperation.

I will take a look at it and i will probably generate some C code that will directly write the files instead of creating the template.

Can you also provide script that generated the NVS variant ? 

I would probably write it so that it supports both LittleFS and NVS.

Additionally do you have some C backend code for the LittleFS somewhere that i can see as a starting point for the C variant ?

peterbornerup commented:
I've added some additional files  (originally added to reduce the writes) 

The NVS partition was brutally created by exporting everything as a big c-array and writing using the NVS on the chip. This could technically probably be done without using a physical device? 

Anyway, after doing that I dumped the flash and picked out the relevant stuff. I've added a python script to generate this kinda c style thing. 

please note that the nvs and littlefs probably never will be 1:1 compatible. In the NVS struct we have to write a mapping between paths and IDs in the the first entry. LittleFS can just use the actual paths :)
